### PR TITLE
Add Sudoku example.

### PR DESCRIPTION
**Pull request by benlaurie**
Just for fun.

### CODE REVIEW

**Comment by wintersteiger** [comment]
Thanks for the example, this looks good to me. 

If it's not too much trouble, could I get you to sign a [CLA](https://github.com/Z3Prover/z3/wiki/Contribution-Guidelines) before we merge this?

**Comment by wintersteiger** [comment]
CLA not required, example code exception.

File: examples/c++/example.cpp [hunk 1]
```diff
@@ -974,6 +974,83 @@ void substitute_example() {
...
+    for (unsigned j = 0; j < 9; ++j) {
+        expr_vector t(c);
+        for (unsigned i = 0; i < 9; ++i)
+            t.push_back(x[i * 9 + j]);
+        s.add(distinct(t));
+    }
+
+    // each 3x3 square contains a digit at most once
+    for (unsigned i0 = 0; i0 < 3; i0++) {
+        for (uint j0 = 0; j0 < 3; j0++) {
```
Comment (benlaurie) [inline examples/c++/example.cpp]: Just noticed: this really should be unsigned, not uint, for consistency.

**Comment by wintersteiger** [comment]
And for it to compile as well! Currently fixing it.

**Pull request merged**